src/main/java/com/example/onlinebookstore/controller/BookController.java

```diff
@@ -29,25 +30,29 @@ public class BookController {
 
     @GetMapping
     @Operation(summary = "Get all books", description = "Get list of all available books")
+    @PreAuthorize("hasAnyRole('ROLE_USER','ROLE_ADMIN')")
```

Comment:
we need to add `ROLE_` when we use `hasAuthorithy()` but here we already specified that we are checking exactly role.
Same for other similar cases
```suggestion
    @PreAuthorize("hasAnyRole('USER','ADMIN')")
```

Reply:
done